pkg/proxy/iptables/proxier_test.go

```diff
@@ -419,17 +419,44 @@ func NewFakeProxier(ipt utiliptables.Interface) *Proxier {
 	return p
 }
 
-func countRules(table, data string) int {
-	inRightTable := false
-	rules := 0
-	for _, line := range strings.Split(data, "\n") {
-		if line == "" {
-			continue
-		}
-		if line[0] == '*' {
-			inRightTable = (line == "*"+table)
+// parseIPTablesData takes iptables-save output and returns a map of table name to array of lines.
+func parseIPTablesData(ruleData string) (map[string][]string, error) {
```

Comment:
this is implicitly tested but ideally we can have an unit test just to cover it against refactors

Reply:
I say this because it is the core of all the other functions and a mistake here can be carried over and hard to notice, I think that for new test we are mostly copy and pasting current test assuming current content is right